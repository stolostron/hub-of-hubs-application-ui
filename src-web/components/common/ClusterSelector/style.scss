/** *****************************************************************************
 * Licensed Materials - Property of IBM
 * (c) Copyright IBM Corporation 2020. All Rights Reserved.
 *
 * US Government Users Restricted Rights - Use, duplication or disclosure
 * restricted by GSA ADP Schedule Contract with IBM Corp.
 *******************************************************************************/
// Copyright (c) 2020 Red Hat, Inc.
// Copyright Contributors to the Open Cluster Management project
@import 'scss/variables';

.clusterSelector-container {
  font-size: 14px;
  position: relative;

  .pf-c-check__label {
    font-size: 14px;
    font-weight: 700;
    color: #152935;
  }

  .pf-c-accordion {
    display: block;
  }

  // Cluster label checkbox
  .clusterSelector-checkbox {
    font-weight: 700;
  }

  // Cluster label description
  .labels-descr {
    font-size: 14px;
    margin-bottom: 30px;
  }

  // Cluster label section
  .labels-section {
    display: block;
  }

  .matching-labels-container {
    display: flex;
    margin-bottom: 20px;
  }

  .matching-labels-input {
    max-width: 45%;
    margin-right: 10px;
    overflow: hidden;
  }

  .text-input {
    background-color: $acm_background_color;
  }

  .add-label-btn {
    width: 190px;
    height: 40px;
    padding: 5px;
    text-align: center;
    cursor: pointer;
  }

  .btn-disabled {
    cursor: not-allowed !important;
  }

  .add-label-btn-icon {
    margin-left: 7px;
    float: left;
  }

  .add-label-btn-text {
    font-weight: 700;
    font-size: 14px;
    color: #06c;
  }

  .remove-label-btn {
    padding: 12px 5px;
    cursor: pointer;
  }
}