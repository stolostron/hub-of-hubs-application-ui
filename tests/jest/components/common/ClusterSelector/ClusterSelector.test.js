/** *****************************************************************************
 * Licensed Materials - Property of IBM
 * (c) Copyright IBM Corporation 2019. All Rights Reserved.
 *
 * Note to U.S. Government Users Restricted Rights:
 * Use, duplication or disclosure restricted by GSA ADP Schedule
 * Contract with IBM Corp.
 ****************************************************************************** */
// Copyright (c) 2020 Red Hat, Inc.
// Copyright Contributors to the Open Cluster Management project

"use strict";

import React from "react";
import ClusterSelector from "../../../../../src-web/components/common/ClusterSelector";
import renderer from "react-test-renderer";
import _ from "lodash";
import { shallow } from "enzyme";

const handleChange = jest.fn((value, targetName, targetID) => {
  return null;
});

const validation = jest.fn(exceptions => {
  return null;
});

const summarizeFn = jest.fn(() => {
  return null;
});

const emptyControl = {
  active: {},
  validation: validation,
  summarize: summarizeFn
};

const emptyControlShowData = {
  active: {},
  showData: [
    {
      id: 0,
      labelName: "name",
      labelValue: "ui-e2e-remote",
      validValue: true
    }
  ],
  validation: validation,
  summarize: summarizeFn
};

const controlActive = {
  active: {
    mode: true,
    clusterLabelsList: [
      {
        id: 0,
        labelName: "env",
        labelValue: "Dev",
        validValue: true
      },
      {
        id: 1,
        labelName: "cloud",
        labelValue: "AWS",
        validValue: true
      }
    ],
    clusterLabelsListID: 2
  },
  validation: validation,
  summarize: summarizeFn
};

describe("ClusterSelector component", () => {
  it("renders full control", () => {
    const component = renderer.create(
      <ClusterSelector
        locale="en-US"
        type="custom"
        available={[]}
        control={controlActive}
        handleChange={handleChange}
      />
    );
    expect(component.toJSON()).toMatchSnapshot();
  });

  it("renders empty control", () => {
    const component = renderer.create(
      <ClusterSelector
        locale="en-US"
        type="custom"
        available={[]}
        control={emptyControl}
        handleChange={handleChange}
      />
    );
    expect(component.toJSON()).toMatchSnapshot();
  });

  it("renders empty control with data set", () => {
    const component = renderer.create(
      <ClusterSelector
        locale="en-US"
        type="custom"
        available={[]}
        control={emptyControlShowData}
        handleChange={handleChange}
      />
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
});

describe("on control change function", () => {
  it("renders active control", () => {
    const wrapper = shallow(
      <ClusterSelector
        locale="en-US"
        type="custom"
        available={[]}
        control={controlActive}
        handleChange={handleChange}
      />
    );
    const evt = {
      target: {
        value: "value-testing"
      },
      selectedItems: ["selectedItems-testing-1", "selectedItems-testing-2"]
    };

    wrapper
      .find("#clusterSelector-checkbox-undefined")
      .at(0)
      .simulate("change", true);
    wrapper
      .find("#labels-header")
      .at(0)
      .simulate("keypress", evt);

    wrapper
      .find("#labels-header")
      .at(0)
      .simulate("click", evt);

    wrapper.find(".add-label-btn").simulate("click", true, controlActive);
    wrapper.find(".add-label-btn").simulate("keypress", { type: "click" });

    wrapper.find("#labelName-0-undefined").simulate("change", true);
    wrapper.find("#labelValue-0-undefined").simulate("change", true);

    //
    wrapper
      .find(".remove-label-btn")
      .at(0)
      .simulate("click", true);
  });
});
