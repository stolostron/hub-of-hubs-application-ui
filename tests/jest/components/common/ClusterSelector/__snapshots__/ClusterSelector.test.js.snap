// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClusterSelector component renders empty control 1`] = `
<div
  className="creation-view-controls-labels"
>
  <div />
  <div
    className="clusterSelector-container"
  >
    <div
      className="pf-c-check clusterSelector-checkbox"
    >
      <input
        aria-invalid={false}
        checked={true}
        className="pf-c-check__input"
        data-ouia-component-id="OUIA-Generated-Checkbox-2"
        data-ouia-component-type="PF4/Checkbox"
        data-ouia-safe={true}
        disabled={false}
        id="clusterSelector-checkbox-undefined"
        onChange={[Function]}
        type="checkbox"
      />
      <label
        className="pf-c-check__label"
        htmlFor="clusterSelector-checkbox-undefined"
      >
        Deploy application resources only on clusters matching specified labels
      </label>
    </div>
    <dl
      aria-label=""
      className="pf-c-accordion"
    >
      <dt>
        <button
          aria-expanded={true}
          className="pf-c-accordion__toggle pf-m-expanded"
          id="labels-header"
          onClick={[Function]}
          type="button"
        >
          <span
            className="pf-c-accordion__toggle-text"
          >
            Cluster labels
          </span>
          <span
            className="pf-c-accordion__toggle-icon"
          >
            <svg
              aria-hidden={true}
              aria-labelledby={null}
              fill="currentColor"
              height="1em"
              role="img"
              style={
                Object {
                  "verticalAlign": "-0.125em",
                }
              }
              viewBox="0 0 256 512"
              width="1em"
            >
              <path
                d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34z"
              />
            </svg>
          </span>
        </button>
      </dt>
      <dd
        aria-label=""
        className="pf-c-accordion__expanded-content pf-m-expanded"
        hidden={false}
        id=""
      >
        <div
          className="pf-c-accordion__expanded-content-body"
        >
          <div
            className="clusterSelector-labels-section"
          >
            <div
              className="labels-descr"
            >
              Enter one or more matching labels to select the clusters to deploy to
            </div>
            <div
              className="labels-section"
              id="clusterSelector-labels-section-undefined"
            >
              <div
                className="matching-labels-container"
              >
                <div
                  className="matching-labels-input"
                >
                  <div
                    className="pf-c-form__group"
                    id="labelName-0-undefined-label"
                  >
                    <div
                      className="pf-c-form__group-label"
                    >
                      <label
                        className="pf-c-form__label"
                        htmlFor="labelName-0-undefined"
                      >
                        <span
                          className="pf-c-form__label-text"
                        >
                          Label
                        </span>
                        <span
                          aria-hidden="true"
                          className="pf-c-form__label-required"
                        >
                           
                          *
                        </span>
                      </label>
                       
                    </div>
                    <div
                      className="pf-c-form__group-control"
                    >
                      <input
                        aria-invalid={false}
                        aria-label={null}
                        className="pf-c-form-control text-input"
                        data-ouia-component-id="OUIA-Generated-TextInputBase-5"
                        data-ouia-component-type="PF4/TextInput"
                        data-ouia-safe={true}
                        disabled={false}
                        id="labelName-0-undefined"
                        label="Label"
                        onBlur={[Function]}
                        onChange={[Function]}
                        onFocus={[Function]}
                        placeholder="Label name"
                        readOnly={false}
                        required={true}
                        type="text"
                        value=""
                      />
                      
                    </div>
                  </div>
                </div>
                <div
                  className="matching-labels-input"
                >
                  <div
                    className="pf-c-form__group"
                    id="labelValue-0-undefined-label"
                  >
                    <div
                      className="pf-c-form__group-label"
                    >
                      <label
                        className="pf-c-form__label"
                        htmlFor="labelValue-0-undefined"
                      >
                        <span
                          className="pf-c-form__label-text"
                        >
                          Value
                        </span>
                        <span
                          aria-hidden="true"
                          className="pf-c-form__label-required"
                        >
                           
                          *
                        </span>
                      </label>
                       
                    </div>
                    <div
                      className="pf-c-form__group-control"
                    >
                      <input
                        aria-invalid={false}
                        aria-label={null}
                        className="pf-c-form-control text-input"
                        data-ouia-component-id="OUIA-Generated-TextInputBase-6"
                        data-ouia-component-type="PF4/TextInput"
                        data-ouia-safe={true}
                        disabled={false}
                        id="labelValue-0-undefined"
                        label="Value"
                        onBlur={[Function]}
                        onChange={[Function]}
                        onFocus={[Function]}
                        placeholder="Label value"
                        readOnly={false}
                        required={true}
                        type="text"
                        value=""
                      />
                      
                    </div>
                  </div>
                </div>
                
              </div>
              <div
                className="add-label-btn "
                onClick={[Function]}
                onKeyPress={[Function]}
                role="button"
                tabIndex="0"
              >
                <svg
                  aria-hidden={true}
                  aria-labelledby={null}
                  className="add-label-btn-icon"
                  fill="#06c"
                  height="1em"
                  role="img"
                  style={
                    Object {
                      "verticalAlign": "-0.125em",
                    }
                  }
                  viewBox="0 0 512 512"
                  width="1em"
                >
                  <path
                    d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm144 276c0 6.6-5.4 12-12 12h-92v92c0 6.6-5.4 12-12 12h-56c-6.6 0-12-5.4-12-12v-92h-92c-6.6 0-12-5.4-12-12v-56c0-6.6 5.4-12 12-12h92v-92c0-6.6 5.4-12 12-12h56c6.6 0 12 5.4 12 12v92h92c6.6 0 12 5.4 12 12v56z"
                  />
                </svg>
                <div
                  className="add-label-btn-text"
                >
                  Add another label
                </div>
              </div>
            </div>
          </div>
        </div>
      </dd>
    </dl>
  </div>
</div>
`;

exports[`ClusterSelector component renders empty control with data set 1`] = `
<div
  className="creation-view-controls-labels"
>
  <div />
  <div
    className="clusterSelector-container"
  >
    <div
      className="pf-c-check clusterSelector-checkbox"
    >
      <input
        aria-invalid={false}
        checked={true}
        className="pf-c-check__input"
        data-ouia-component-id="OUIA-Generated-Checkbox-3"
        data-ouia-component-type="PF4/Checkbox"
        data-ouia-safe={true}
        disabled={true}
        id="clusterSelector-checkbox-undefined"
        onChange={[Function]}
        type="checkbox"
      />
      <label
        className="pf-c-check__label pf-m-disabled"
        htmlFor="clusterSelector-checkbox-undefined"
      >
        Deploy application resources only on clusters matching specified labels
      </label>
    </div>
    <dl
      aria-label=""
      className="pf-c-accordion"
    >
      <dt>
        <button
          aria-expanded={true}
          className="pf-c-accordion__toggle pf-m-expanded"
          id="labels-header"
          onClick={[Function]}
          type="button"
        >
          <span
            className="pf-c-accordion__toggle-text"
          >
            Cluster labels
          </span>
          <span
            className="pf-c-accordion__toggle-icon"
          >
            <svg
              aria-hidden={true}
              aria-labelledby={null}
              fill="currentColor"
              height="1em"
              role="img"
              style={
                Object {
                  "verticalAlign": "-0.125em",
                }
              }
              viewBox="0 0 256 512"
              width="1em"
            >
              <path
                d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34z"
              />
            </svg>
          </span>
        </button>
      </dt>
      <dd
        aria-label=""
        className="pf-c-accordion__expanded-content pf-m-expanded"
        hidden={false}
        id=""
      >
        <div
          className="pf-c-accordion__expanded-content-body"
        >
          <div
            className="clusterSelector-labels-section"
          >
            <div
              className="labels-descr"
            >
              Enter one or more matching labels to select the clusters to deploy to
            </div>
            <div
              className="labels-section"
              id="clusterSelector-labels-section-undefined"
            >
              <div
                className="matching-labels-container"
              >
                <div
                  className="matching-labels-input"
                >
                  <div
                    className="pf-c-form__group"
                    id="labelName-0-undefined-label"
                  >
                    <div
                      className="pf-c-form__group-label"
                    >
                      <label
                        className="pf-c-form__label"
                        htmlFor="labelName-0-undefined"
                      >
                        <span
                          className="pf-c-form__label-text"
                        >
                          Label
                        </span>
                        <span
                          aria-hidden="true"
                          className="pf-c-form__label-required"
                        >
                           
                          *
                        </span>
                      </label>
                       
                    </div>
                    <div
                      className="pf-c-form__group-control"
                    >
                      <input
                        aria-invalid={false}
                        aria-label={null}
                        className="pf-c-form-control text-input"
                        data-ouia-component-id="OUIA-Generated-TextInputBase-7"
                        data-ouia-component-type="PF4/TextInput"
                        data-ouia-safe={true}
                        disabled={true}
                        id="labelName-0-undefined"
                        label="Label"
                        onBlur={[Function]}
                        onChange={[Function]}
                        onFocus={[Function]}
                        placeholder="Label name"
                        readOnly={false}
                        required={true}
                        type="text"
                        value="name"
                      />
                      
                    </div>
                  </div>
                </div>
                <div
                  className="matching-labels-input"
                >
                  <div
                    className="pf-c-form__group"
                    id="labelValue-0-undefined-label"
                  >
                    <div
                      className="pf-c-form__group-label"
                    >
                      <label
                        className="pf-c-form__label"
                        htmlFor="labelValue-0-undefined"
                      >
                        <span
                          className="pf-c-form__label-text"
                        >
                          Value
                        </span>
                        <span
                          aria-hidden="true"
                          className="pf-c-form__label-required"
                        >
                           
                          *
                        </span>
                      </label>
                       
                    </div>
                    <div
                      className="pf-c-form__group-control"
                    >
                      <input
                        aria-invalid={false}
                        aria-label={null}
                        className="pf-c-form-control text-input"
                        data-ouia-component-id="OUIA-Generated-TextInputBase-8"
                        data-ouia-component-type="PF4/TextInput"
                        data-ouia-safe={true}
                        disabled={true}
                        id="labelValue-0-undefined"
                        label="Value"
                        onBlur={[Function]}
                        onChange={[Function]}
                        onFocus={[Function]}
                        placeholder="Label value"
                        readOnly={false}
                        required={true}
                        type="text"
                        value="ui-e2e-remote"
                      />
                      
                    </div>
                  </div>
                </div>
                
              </div>
              <div
                className="add-label-btn btn-disabled"
                onClick={[Function]}
                onKeyPress={[Function]}
                role="button"
                tabIndex="0"
              >
                <svg
                  aria-hidden={true}
                  aria-labelledby={null}
                  className="add-label-btn-icon"
                  fill="#06c"
                  height="1em"
                  role="img"
                  style={
                    Object {
                      "verticalAlign": "-0.125em",
                    }
                  }
                  viewBox="0 0 512 512"
                  width="1em"
                >
                  <path
                    d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm144 276c0 6.6-5.4 12-12 12h-92v92c0 6.6-5.4 12-12 12h-56c-6.6 0-12-5.4-12-12v-92h-92c-6.6 0-12-5.4-12-12v-56c0-6.6 5.4-12 12-12h92v-92c0-6.6 5.4-12 12-12h56c6.6 0 12 5.4 12 12v92h92c6.6 0 12 5.4 12 12v56z"
                  />
                </svg>
                <div
                  className="add-label-btn-text"
                >
                  Add another label
                </div>
              </div>
            </div>
          </div>
        </div>
      </dd>
    </dl>
  </div>
</div>
`;

exports[`ClusterSelector component renders full control 1`] = `
<div
  className="creation-view-controls-labels"
>
  <div />
  <div
    className="clusterSelector-container"
  >
    <div
      className="pf-c-check clusterSelector-checkbox"
    >
      <input
        aria-invalid={false}
        checked={true}
        className="pf-c-check__input"
        data-ouia-component-id="OUIA-Generated-Checkbox-1"
        data-ouia-component-type="PF4/Checkbox"
        data-ouia-safe={true}
        disabled={false}
        id="clusterSelector-checkbox-undefined"
        onChange={[Function]}
        type="checkbox"
      />
      <label
        className="pf-c-check__label"
        htmlFor="clusterSelector-checkbox-undefined"
      >
        Deploy application resources only on clusters matching specified labels
      </label>
    </div>
    <dl
      aria-label=""
      className="pf-c-accordion"
    >
      <dt>
        <button
          aria-expanded={true}
          className="pf-c-accordion__toggle pf-m-expanded"
          id="labels-header"
          onClick={[Function]}
          type="button"
        >
          <span
            className="pf-c-accordion__toggle-text"
          >
            Cluster labels
          </span>
          <span
            className="pf-c-accordion__toggle-icon"
          >
            <svg
              aria-hidden={true}
              aria-labelledby={null}
              fill="currentColor"
              height="1em"
              role="img"
              style={
                Object {
                  "verticalAlign": "-0.125em",
                }
              }
              viewBox="0 0 256 512"
              width="1em"
            >
              <path
                d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34z"
              />
            </svg>
          </span>
        </button>
      </dt>
      <dd
        aria-label=""
        className="pf-c-accordion__expanded-content pf-m-expanded"
        hidden={false}
        id=""
      >
        <div
          className="pf-c-accordion__expanded-content-body"
        >
          <div
            className="clusterSelector-labels-section"
          >
            <div
              className="labels-descr"
            >
              Enter one or more matching labels to select the clusters to deploy to
            </div>
            <div
              className="labels-section"
              id="clusterSelector-labels-section-undefined"
            >
              <div
                className="matching-labels-container"
              >
                <div
                  className="matching-labels-input"
                >
                  <div
                    className="pf-c-form__group"
                    id="labelName-0-undefined-label"
                  >
                    <div
                      className="pf-c-form__group-label"
                    >
                      <label
                        className="pf-c-form__label"
                        htmlFor="labelName-0-undefined"
                      >
                        <span
                          className="pf-c-form__label-text"
                        >
                          Label
                        </span>
                        <span
                          aria-hidden="true"
                          className="pf-c-form__label-required"
                        >
                           
                          *
                        </span>
                      </label>
                       
                    </div>
                    <div
                      className="pf-c-form__group-control"
                    >
                      <input
                        aria-invalid={false}
                        aria-label={null}
                        className="pf-c-form-control text-input"
                        data-ouia-component-id="OUIA-Generated-TextInputBase-1"
                        data-ouia-component-type="PF4/TextInput"
                        data-ouia-safe={true}
                        disabled={false}
                        id="labelName-0-undefined"
                        label="Label"
                        onBlur={[Function]}
                        onChange={[Function]}
                        onFocus={[Function]}
                        placeholder="Label name"
                        readOnly={false}
                        required={true}
                        type="text"
                        value="env"
                      />
                      
                    </div>
                  </div>
                </div>
                <div
                  className="matching-labels-input"
                >
                  <div
                    className="pf-c-form__group"
                    id="labelValue-0-undefined-label"
                  >
                    <div
                      className="pf-c-form__group-label"
                    >
                      <label
                        className="pf-c-form__label"
                        htmlFor="labelValue-0-undefined"
                      >
                        <span
                          className="pf-c-form__label-text"
                        >
                          Value
                        </span>
                        <span
                          aria-hidden="true"
                          className="pf-c-form__label-required"
                        >
                           
                          *
                        </span>
                      </label>
                       
                    </div>
                    <div
                      className="pf-c-form__group-control"
                    >
                      <input
                        aria-invalid={false}
                        aria-label={null}
                        className="pf-c-form-control text-input"
                        data-ouia-component-id="OUIA-Generated-TextInputBase-2"
                        data-ouia-component-type="PF4/TextInput"
                        data-ouia-safe={true}
                        disabled={false}
                        id="labelValue-0-undefined"
                        label="Value"
                        onBlur={[Function]}
                        onChange={[Function]}
                        onFocus={[Function]}
                        placeholder="Label value"
                        readOnly={false}
                        required={true}
                        type="text"
                        value="Dev"
                      />
                      
                    </div>
                  </div>
                </div>
                
              </div>
              <div
                className="matching-labels-container"
              >
                <div
                  className="matching-labels-input"
                >
                  <div
                    className="pf-c-form__group"
                    id="labelName-1-undefined-label"
                  >
                    
                    <div
                      className="pf-c-form__group-control"
                    >
                      <input
                        aria-invalid={false}
                        aria-label={null}
                        className="pf-c-form-control text-input"
                        data-ouia-component-id="OUIA-Generated-TextInputBase-3"
                        data-ouia-component-type="PF4/TextInput"
                        data-ouia-safe={true}
                        disabled={false}
                        id="labelName-1-undefined"
                        label=""
                        onBlur={[Function]}
                        onChange={[Function]}
                        onFocus={[Function]}
                        placeholder="Label name"
                        readOnly={false}
                        required={true}
                        type="text"
                        value="cloud"
                      />
                      
                    </div>
                  </div>
                </div>
                <div
                  className="matching-labels-input"
                >
                  <div
                    className="pf-c-form__group"
                    id="labelValue-1-undefined-label"
                  >
                    
                    <div
                      className="pf-c-form__group-control"
                    >
                      <input
                        aria-invalid={false}
                        aria-label={null}
                        className="pf-c-form-control text-input"
                        data-ouia-component-id="OUIA-Generated-TextInputBase-4"
                        data-ouia-component-type="PF4/TextInput"
                        data-ouia-safe={true}
                        disabled={false}
                        id="labelValue-1-undefined"
                        label=""
                        onBlur={[Function]}
                        onChange={[Function]}
                        onFocus={[Function]}
                        placeholder="Label value"
                        readOnly={false}
                        required={true}
                        type="text"
                        value="AWS"
                      />
                      
                    </div>
                  </div>
                </div>
                <div
                  className="remove-label-btn "
                  id={1}
                  onClick={[Function]}
                  onKeyPress={[Function]}
                  role="button"
                  tabIndex="0"
                >
                  <svg
                    aria-hidden={true}
                    aria-labelledby={null}
                    fill="#06c"
                    height="1em"
                    role="img"
                    style={
                      Object {
                        "verticalAlign": "-0.125em",
                      }
                    }
                    viewBox="0 0 512 512"
                    width="1em"
                  >
                    <path
                      d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm121.6 313.1c4.7 4.7 4.7 12.3 0 17L338 377.6c-4.7 4.7-12.3 4.7-17 0L256 312l-65.1 65.6c-4.7 4.7-12.3 4.7-17 0L134.4 338c-4.7-4.7-4.7-12.3 0-17l65.6-65-65.6-65.1c-4.7-4.7-4.7-12.3 0-17l39.6-39.6c4.7-4.7 12.3-4.7 17 0l65 65.7 65.1-65.6c4.7-4.7 12.3-4.7 17 0l39.6 39.6c4.7 4.7 4.7 12.3 0 17L312 256l65.6 65.1z"
                    />
                  </svg>
                </div>
              </div>
              <div
                className="add-label-btn "
                onClick={[Function]}
                onKeyPress={[Function]}
                role="button"
                tabIndex="0"
              >
                <svg
                  aria-hidden={true}
                  aria-labelledby={null}
                  className="add-label-btn-icon"
                  fill="#06c"
                  height="1em"
                  role="img"
                  style={
                    Object {
                      "verticalAlign": "-0.125em",
                    }
                  }
                  viewBox="0 0 512 512"
                  width="1em"
                >
                  <path
                    d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm144 276c0 6.6-5.4 12-12 12h-92v92c0 6.6-5.4 12-12 12h-56c-6.6 0-12-5.4-12-12v-92h-92c-6.6 0-12-5.4-12-12v-56c0-6.6 5.4-12 12-12h92v-92c0-6.6 5.4-12 12-12h56c6.6 0 12 5.4 12 12v92h92c6.6 0 12 5.4 12 12v56z"
                  />
                </svg>
                <div
                  className="add-label-btn-text"
                >
                  Add another label
                </div>
              </div>
            </div>
          </div>
        </div>
      </dd>
    </dl>
  </div>
</div>
`;
